noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -156,8 +156,11 @@ pub(crate) comptime fn generate_note_export(
     fields: [(Quoted, u32, bool)]
 ) -> Quoted {
     let name = s.name();
-    let global_export_name = f"{name}_EXPORTS".quoted_contents();
-    let note_fields_name = f"{name}Fields".quoted_contents();
+    let mut hasher = Poseidon2Hasher::default();
+    s.as_type().hash(&mut hasher);
+    let hash = hasher.finish() as u32;
+    let global_export_name = f"{name}_{hash}_EXPORTS".quoted_contents();
+    let note_fields_name = f"{name}Fields_{hash}".quoted_contents();
```

Comment:
With my work on having the note fields and indices in the ABI  there is now NoteFields exported and I got collision here as well so I had to do the same change as what Grego advised me to do for the exports.